### PR TITLE
imp: add possibility to inject authorisation header to services via RequestData

### PR DESCRIPTION
**Pull request by akomisarek**
### Change description ###

As in title - you can Inject type RequestData to your services and as long as you are in the request (NOT ASYNC processing) you will get authorization header. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class AuthorizationConfiguration {
+
+    @Bean
+    @RequestScope
+    public RequestData authorisation(HttpServletRequest request) {
+        String authorization = request.getHeader("authorization");
+
+        return new RequestData(authorization);
+    }
+}
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java]: request data does not have to be request scoped, by default injected HttpServletRequest is request scoped.
can be just:
@Service
public class RequestData {
    @Autowired
    HttpServletRequest httpServletRequest;

    public String authorsation() {
        return httpServletRequest.getHeader("authorization");
    }
}

File: service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.web.context.annotation.RequestScope;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Configuration
+public class AuthorizationConfiguration {
+
+    @Bean
+    @RequestScope
+    public RequestData authorisation(HttpServletRequest request) {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java]: class and method is about authorisation, but returns quite generic RequestData

Comment (Gogetter) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java]: I think this might be expanded to include `user-id`

Comment (akomisarek) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java]: Refactored to what we discussed, thanks

File: service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java (hunk 1 above)
Comment (akomisarek) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/request/AuthorizationConfiguration.java]: Thanks, changing!

**Pull request merged**